pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/minion/PinotHelixTaskResourceManager.java

```diff
@@ -127,67 +163,108 @@ public String submitTask(@Nonnull PinotTaskConfig pinotTaskConfig, @Nonnull Stri
+        _taskDriver.getJobConfig(getHelixJobName(taskName)).getTaskConfig(taskName));
   }
 
   /**
    * Helper method to convert task type to Helix JobQueue name.
    * <p>E.g. DummyTask -> TaskQueue_DummyTask
+   *
+   * @param taskType Task type
+   * @return Helix JobQueue name
    */
   @Nonnull
   protected static String getHelixJobQueueName(@Nonnull String taskType) {
     return TASK_QUEUE_PREFIX + taskType;
   }
 
   /**
-   * Helper method to convert Pinot task name to Helix Job name with JobQueue prefix based on the given task type.
-   * <p>E.g. Task_DummyTask_12345 (type DummyTask) -> TaskQueue_DummyTask_Task_DummyTask_12345
+   * Helper method to convert Pinot task name to Helix Job name with JobQueue prefix.
+   * <p>E.g. Task_DummyTask_12345 -> TaskQueue_DummyTask_Task_DummyTask_12345
+   *
+   * @param pinotTaskName Pinot task name
+   * @return helixJobName Helix Job name
    */
   @Nonnull
-  private static String getHelixJobName(@Nonnull String taskType, @Nonnull String pinotTaskName) {
-    return TASK_QUEUE_PREFIX + taskType + TASK_NAME_SEPARATOR + pinotTaskName;
+  private static String getHelixJobName(@Nonnull String pinotTaskName) {
+    return getHelixJobQueueName(getTaskType(pinotTaskName)) + TASK_NAME_SEPARATOR + pinotTaskName;
   }
 
   /**
-   * Helper method to convert Helix Job name with JobQueue prefix to Pinot task name based on the given task type.
-   * <p>E.g. TaskQueue_DummyTask_Task_DummyTask_12345 (type DummyTask) -> Task_DummyTask_12345
+   * Helper method to convert Helix Job name with JobQueue prefix to Pinot task name.
+   * <p>E.g. TaskQueue_DummyTask_Task_DummyTask_12345 -> Task_DummyTask_12345
+   *
+   * @param helixJobName Helix Job name
+   * @return Pinot task name
    */
   @Nonnull
-  private static String getPinotTaskName(@Nonnull String taskType, @Nonnull String helixJobName) {
-    return helixJobName.substring(TASK_QUEUE_PREFIX.length() + taskType.length() + 1);
+  private static String getPinotTaskName(@Nonnull String helixJobName) {
+    return helixJobName.substring(TASK_QUEUE_PREFIX.length() + getTaskType(helixJobName).length() + 1);
   }
 
   /**
-   * Helper method to extract task type from Pinot task name.
-   * <p>E.g. Task_DummyTask_12345 -> DummyTask
+   * Helper method to extract task type from Pinot task name, Helix JobQueue name or Helix Job name.
+   * <p>E.g. Task_DummyTask_12345 -> DummyTask (from Pinot task name)
+   * <p>E.g. TaskQueue_DummyTask -> DummyTask (from Helix JobQueue name)
+   * <p>E.g. TaskQueue_DummyTask_Task_DummyTask_12345 -> DummyTask (from Helix Job name)
+   *
+   * @param name Pinot task name, Helix JobQueue name or Helix Job name
+   * @return Task type
    */
   @Nonnull
-  private static String getTaskType(@Nonnull String pinotTaskName) {
-    return pinotTaskName.split(TASK_NAME_SEPARATOR)[1];
+  private static String getTaskType(@Nonnull String name) {
+    return name.split(TASK_NAME_SEPARATOR)[1];
```

Comment:
If the naming convention get more complex in the future, I will wrap them into a separate class.